drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
 static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
 {
     at86rf2xx_t *dev = (at86rf2xx_t *)netdev;
-    uint8_t phr;
     size_t pkt_len;
 
+#ifndef MODULE_AT86RFR2
+    uint8_t phr;
+#endif
+
     /* frame buffer protection will be unlocked as soon as at86rf2xx_fb_stop()
      * is called*/
     at86rf2xx_fb_start(dev);
 
     /* get the size of the received packet */
+#ifdef MODULE_AT86RFR2
+    /* Transceiver Received Frame Length Register (refer p.35, 85, 154) */
+    /* subtract length of FCS */
+    pkt_len = (TST_RX_LENGTH & 0x7f) - 2;
+#else
     at86rf2xx_fb_read(dev, &phr, 1);
 
-    /* ignore MSB (refer p.80) and substract length of FCS field */
+    /* ignore MSB (refer p.80) and subtract length of FCS field */
     pkt_len = (phr & 0x7f) - 2;
+#endif
 
-    /* just return length when buf == NULL */
+    /* just return length when buf == NULL, Upper layer calls _recv
+     * with no buffer to get expected size for allocating buffer
+     *
+     * Attetion:fb_stop does remove the frame buffer protection !
+     * */
     if (buf == NULL) {
+        /* TODO Is this not a problem? Frame Buffer release */
         at86rf2xx_fb_stop(dev);
+        DEBUG("Packet lenght: %d\n", pkt_len);
         return pkt_len;
     }
     /* not enough space in buf */
     if (pkt_len > len) {
         at86rf2xx_fb_stop(dev);
+#ifdef MODULE_AT86RFR2
+        /* If Upper layer does not supply a big enough buffer data loss is
+         * unavoidable clear frame buffer protection
+         * */
+        *AT86RF2XX_REG__TRX_CTRL_2 &= ~(1 << RX_SAFE_MODE);
```

Comment:
You disable the FB protection here, but you also enable it for any device. I think this may cause conflicts...

Reply:
Normally `at86rf2xx_fb_stop(dev);` would release the frame buffer, but in this implementation i do not release the frame buffer in `at86rf2xx_fb_stop(dev);`, so it has to be done additionally.
And it is in a `#ifdef MODULE_AT86RFR2`.